### PR TITLE
ORC-1443: Enforce Java version

### PR DESCRIPTION
**Pull request by williamhyun**
### What changes were proposed in this pull request?
This PR aims to add enforce java version feature.
At this moment it is enforced to 1.8, however, we will be able to move it up later. 

### Why are the changes needed?
This feature helps enforce the minimum java version.

### How was this patch tested?
Pass the CIs.

### CODE REVIEW

**Comment by williamhyun** [comment]
@dongjoon-hyun

**Review by dongjoon-hyun — APPROVED** [review]
Thank you for pinging me. This looks good for v1.9.0.

File: java/pom.xml [hunk 1]
```diff
@@ -60,6 +60,7 @@
   </modules>
 
   <properties>
+    <java.version>1.8</java.version>
```
Comment (dongjoon-hyun) [inline java/pom.xml:63]: Oh, could you put this new property in a sorted order? CI seems to complain at the order.

Comment (dongjoon-hyun) [reply java/pom.xml]: Let me fix this~

**Comment by williamhyun** [comment]
Thank you for the review @dongjoon-hyun !

**Comment by dongjoon-hyun** [comment]
Thank you! I cherry-picked this to branch-1.9, too.

**Pull request closed without merging**